### ISSUE TITLE
RTL app direction

### ISSUE THREAD

**Issue by SayWut** [subhamayd2/day_night_time_picker#10]
OS: android
OS Version: 8.0
Device: oneplus 3T

The picked time in RTL app layout isn't display right

instead of displaying:
`HH:MM` (24 hours format)
`H:MM` (12 hours format)

it displaying:
`MM:HH` (24 hours format)
`MM:H` (12 hours format)

![1](https://user-images.githubusercontent.com/36565977/99845783-e3577c80-2b7d-11eb-9a08-f786c35546a8.jpeg)
![2](https://user-images.githubusercontent.com/36565977/99845785-e3f01300-2b7d-11eb-9a0c-105458c93b83.jpeg)

UPDATE:
So this is my solution for this problem until it will be fixed:

```dart
Navigator.of(context).push(
  PageRouteBuilder(
	pageBuilder: (context, _, __) {
	  return Directionality(
		textDirection: TextDirection.ltr,
		child: showPicker(
		  value: value,
		  onChange: onChanged,
		  is24HrFormat: true,
		).buildPage(context, _, __),
	  );
	},
	transitionDuration: Duration(milliseconds: 200),
	transitionsBuilder: (context, anim, secondAnim, child) => SlideTransition(
	  position: anim.drive(
		Tween(
		  begin: const Offset(0, 0.15),
		  end: const Offset(0, 0),
		).chain(
		  CurveTween(curve: Curves.ease),
		),
	  ),
	  child: FadeTransition(
		opacity: anim,
		child: child,
	  ),
	),
	barrierDismissible: true,
	opaque: false,
	barrierColor: Colors.black45,
  ),
```

basically I just wrapped the time pick dialog with `PageRouteBuilder` and `Directionality` and I set the `textDirection` property to LRT.
The rest of the `PageRouteBuilder` properties is the same of the original time pick dialog, I just copied it to maintain the same look and feel of the original dialog

Sorry for bad english

**Comment by subhamayd2** [comment]
Hi,

Fixed the issue in the new release.
Please let me know, in case of any problems.

Also, I could not check with a global `textDirection` for the app as a whole. For some reason, even when I wrapped the root widget with `Directionality()` widget, the popup was not setting itself to be `RTL`.

Kindly check on your end and let me know.

Thanks.

**Comment by SayWut** [comment]
> Hi,
> 
> Fixed the issue in the new release.
> Please let me know, in case of any problems.
> 
> Also, I could not check with a global `textDirection` for the app as a whole. For some reason, even when I wrapped the root widget with `Directionality()` widget, the popup was not setting itself to be `RTL`.
> 
> Kindly check on your end and let me know.
> 
> Thanks.

The text is fixed but the slider is going from right to left intead from left to right
I edited your code on my end and it works like a charm:

```dart
Directionality(
  textDirection: TextDirection.ltr,
  child: Slider(
	value: changingHour ? hour.roundToDouble() : minute.roundToDouble(),
	onChanged: onChangeTime,
	min: min,
	max: max,
	divisions: divisions,
	activeColor: color,
	inactiveColor: color.withAlpha(55),
  ),
),
```

**Comment by subhamayd2** [comment]
@SayWut but for RTL layout shouldn't the slider be from right-to-left too?

**Comment by SayWut** [comment]
@subhamayd2 idk it feels to me more netural when it is LTR even when my app is RTL
if you want you can add a propery for the slider direction and the default will be `Directionality.of(context)`

**Comment by subhamayd2** [comment]
That sounds perfect.

**Comment by TheGlorySaint** [comment]
Hi @SayWut,

I have added a bool ltrMode(standard true) that can be set to false for the showPicker and createInlinePicker for iOs and Default Style.

After approval, you can see the changes inside this Dependcie.
